android/src/main/java/com/reactnativearviewer/ArViewerView.kt

```diff
@@ -54,59 +72,346 @@ open class ArViewerView @JvmOverloads constructor(
   /**
    * Set of allowed model transformations (rotate, scale, translate...)
    */
-  private var allowTransform = mutableSetOf<EditableTransform>()
+  private var allowTransform = mutableSetOf<String>()
+
+
+  init {
+    if (checkIsSupportedDevice(context.currentActivity!!)) {
+      // let's create sceneform view
+      arView = ArSceneView(context, attrs)
+      arView!!.layoutParams = LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+      this.addView(arView)
+
+      transformationSystem = makeTransformationSystem()
+
+      gestureDetector = GestureDetector(
+        context,
+        object : SimpleOnGestureListener() {
+          override fun onSingleTapUp(e: MotionEvent): Boolean {
+            onSingleTap(e)
+            return true
+          }
+
+          override fun onDown(e: MotionEvent): Boolean {
+            return true
+          }
+        })
+
+      arView!!.scene.addOnPeekTouchListener(this)
+      arView!!.scene.addOnUpdateListener(this)
+      arView!!.viewTreeObserver.addOnWindowFocusChangeListener(onFocusListener)
+      arView!!.setOnSessionConfigChangeListener(this::onSessionConfigChanged)
+
+      val session = Session(context)
+      val config = Config(session)
+
+      // Set plane orientation mode
+      updatePlaneDetection(config)
+      // Enable or not instant placement
+      updateInstantPlacement(config)
+      // Enable or not light estimation
+      updateLightEstimation(config)
+      // Enable or not depth management
+      updateDepthManagement(config)
+
+      config.focusMode = Config.FocusMode.AUTO
+      // Force the non-blocking mode for the session.
+      config.updateMode = Config.UpdateMode.LATEST_CAMERA_IMAGE
+      sessionConfig = config
+      arSession = session
+
+      arView!!.session?.configure(sessionConfig)
+      arView!!.session = arSession
+
+      initializeSession()
+      resume()
+
+      // Setup the instructions view.
+      instructionsController = InstructionsController(context, this);
+      instructionsController!!.setEnabled(true);
+    }
+  }
+
+  private fun resume() {
+    if (isStarted) {
+      return
+    }
+    if ((context as ThemedReactContext).currentActivity != null) {
+      isStarted = true
+      try {
+        arView!!.resume()
+      } catch (ex: java.lang.Exception) {
+        sessionInitializationFailed = true
+      }
+      if (!sessionInitializationFailed) {
+        instructionsController?.setVisible(true)
```

Comment:
It's already on line 131 no?

Reply:
setVisible() is different from setEnable()
If i understound well:
setEnable() activate or not the view
setVisible() display or not the image